greedo-layout/src/main/java/com/fivehundredpx/greedolayout/GreedoLayoutSizeCalculator.java

```diff
@@ -96,34 +109,97 @@ public void computeChildSizesUpToPosition(int lastPosition) {
+                if (mIsFixedHeight) {
+                    itemSlacks = distributeRowSlack(currentRowWidth, rowChildCount, itemAspectRatios);
+
+                    if (!hasValidItemSlacks(itemSlacks, itemAspectRatios)) {
+                        int lastItemWidth = calculateWidth(currentRowHeight,
+                                itemAspectRatios.get(itemAspectRatios.size() - 1));
+                        currentRowWidth -= lastItemWidth;
+                        rowChildCount -= 1;
+                        itemAspectRatios.remove(itemAspectRatios.size() - 1);
+
+                        itemSlacks = distributeRowSlack(currentRowWidth, rowChildCount, itemAspectRatios);
+                    }
+                }
+
                 int availableSpace = mContentWidth;
-                for (Double aspectRatio : aspectRatiosForRow) {
-                    int scaledPhotoWidth = (int)Math.ceil(currentRowHeight * aspectRatio);
-                    scaledPhotoWidth = Math.min(availableSpace, scaledPhotoWidth);
+                for (int i = 0; i < rowChildCount; i++) {
+                    int itemWidth = calculateWidth(currentRowHeight, itemAspectRatios.get(i)) - itemSlacks[i];
+                    itemWidth = Math.min(availableSpace, itemWidth);
 
-                    mSizeForChildAtPosition.add(new Size(scaledPhotoWidth, currentRowHeight));
+                    mSizeForChildAtPosition.add(new Size(itemWidth, currentRowHeight));
                     mRowForChildPosition.add(row);
 
-                    availableSpace = availableSpace - scaledPhotoWidth;
+                    availableSpace -= itemWidth;
                 }
 
-                aspectRatiosForRow.clear();
-                rowAspectRatio = 0.0;
+                itemAspectRatios.clear();
+                currentRowAspectRatio = 0.0;
                 row++;
             }
+
+            pos++;
+        }
+    }
+
+    private int[] distributeRowSlack(int rowWidth, int rowChildCount, List<Double> itemAspectRatios) {
+        return distributeRowSlack(rowWidth - mContentWidth, rowWidth, rowChildCount, itemAspectRatios);
+    }
+
+    private int[] distributeRowSlack(int rowSlack, int rowWidth, int rowChildCount, List<Double> itemAspectRatios) {
+        int itemSlacks[] = new int[rowChildCount];
+
+        for (int i = 0; i < rowChildCount; i++) {
+            double itemWidth = mMaxRowHeight * itemAspectRatios.get(i);
+            itemSlacks[i] = (int) (rowSlack * (itemWidth / rowWidth));
         }
+
+        return itemSlacks;
+    }
+
+    private boolean hasValidItemSlacks(int[] itemSlacks, List<Double> itemAspectRatios) {
+        for (int i = 0; i < itemSlacks.length; i++) {
+            int itemWidth = (int) (itemAspectRatios.get(i) * mMaxRowHeight);
+            if (!isValidItemSlack(itemSlacks[i], itemWidth)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private boolean isValidItemSlack(int itemSlack, int itemWidth) {
```

Comment:
Sorry if I'm misunderstanding, but what about calling this something like `isItemSlackBeyondThreshold`? In that case then, `hasValidItemSlacks()` could be `hasItemSlacksBeyondThreshold()`